### PR TITLE
New numeric test

### PR DESCRIPTION
**Pull request by acgtyrant**
I rewrite the code while referencing `pytorch/aten/src/THNN/generic/BatchNormalization.c`

But it still can not pass two tests:

```
        self.assertTensorClose(input1.grad, input2.grad)
        self.assertTensorClose(batchnorm1.weight.grad, batchnorm2.weight.grad)
```

### CODE REVIEW

**Comment by vacancy** [comment]
Any ideas about why we still can't pass the tests? Is it just numeric issues or probably some implementation error?

**Comment by acgtyrant** [comment]
@vacancy I am not sure, and I do not have many time to solve this, sorry.
